[processing] Add missing resampling methods to GDAL warp.

radosuav opened the pull request:
## Description
Add missing resampling methods to GDAL warp. They are all available since at least GDAL 2.0.0.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ X] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ X] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

pblottiere committed:
Increase cache size

pblottiere committed:
Do not search data in case of an invalid feature

pblottiere committed:
Update constraints when foreign key is manually updated with empty string

pblottiere committed:
Update widget wrapper status only when necessary

borysiasty committed:
[processing] Fix Frequency_analysis' parent group

pblottiere committed:
Reduce functionality in case of a widget already embedded by the same relation

DelazJ committed:
Fix some french translation

radosuav committed:
[processing] Add missing resampling methods to GDAL warp.

borysiasty committed:
Fix a tooltip (backport from 3.0). Fixes #17578

elpaso committed:
[bugfix] Various Bookmarks issues backported

Backport of several master commits that
fix a series of issues in the bookmarks
panel

borysiasty committed:
[tr] Polish translation update

borysiasty committed:
[tr] Polish translation update

borysiasty committed:
[tr] Polish translation update: don't redirect to non-existing Polish documentation

gioman commented:
@radosuav thanks! I had it on my todo list since long. I guess this can be also forward-ported to master.

radosuav commented:
No worries! It looks like all methods are already available on master: https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/warp.py#L69-L80

m-kuhn committed:
[offline] Apply attribute changes on multiple layers

Fix #17647

jef-n committed:
Release of 2.18.15

cjmayo committed:
Fix bookmarks import and export menu items

Method names were changed by 89d52e5d426717cc ("[bugfix] Various
Bookmarks issues backported").

kervel committed:
Fix missing signal/slot connection resulting in non-woring autocompleter for value relation fields (fixes #16676). (#5821)

Backported from d7b08c78faeda9b35882f89913990a253d3b1ab8

jef-n committed:
spelling fixes

jef-n committed:
german translation update

volaya committed:
[processing] added search and replace functionality to script editor

alexbruy committed:
fix indentation

borysiasty committed:
[tr] Polish translation update

pblottiere committed:
Add cacheSize as a QSettings

pblottiere committed:
[bugfix]  Do not add binding line in both side in reshape map tool

pblottiere committed:
Add some tests

pblottiere committed:
Fix indentation

pblottiere committed:
Fixes auto select foreign key when filters are updated

pblottiere committed:
Add some tests

jef-n committed:
2.18 lives in pre-C++11 land (followup c71e55036)

jef-n committed:
2.18 also uses Qt4 which doesn't have QStringLiteral (followup a122e023 and c71e55036)

tomkralidis committed:
[bugfix] [MetaSearch] make gml:Envelope CRS explicit for spatial queries (fixes #17739)

stevenmizuno committed:
change mModel to mMergedModel

to better reflect the purpose; fix object deletion

stevenmizuno committed:
fix export

stevenmizuno committed:
fix bookmark deletion

3nids committed:
[date widget] fix current date can't be picked

better solution has been found for Qt5 but can't be used in Qt4
see #16579

stevenmizuno committed:
refresh the list properly

jef-n committed:
fetch version.txt from version.qgis.org instead of ubuntu.qgis.org
(fixes #17774)

3nids committed:
[date time widget] do not use minimumDateTime for NULL values

stevenmizuno committed:
forgot that this is not 'master'

3nids committed:
fix date/time widget does not handle time zones

issue #16657
manually picked from QGIS3 bdf744ee3f94b5fc1a84e775fedc60a75ba45dcb

jef-n committed:
debian packaging: use libqscintilla2-qt4-dev on unstable

jef-n committed:
2.18 is pre-C++-11 (followup 8e292dff8)

m-kuhn committed:
Fix processing models with finnish translation

Reported in https://gis.stackexchange.com/questions/266819/error-with-graphical-modeler-qgis-not-all-arguments-converted-during-string-for

3nids committed:
use appropriate field type in date time edit widget wrapper

3nids committed:
fix warning

jef-n committed:
date time edit: reapply 718581ff reverted by a8e292df

elpaso committed:
[bugfix][ogr] Recompute capabilities when subsetfilter is set

Backported from master PR 6000

Fixes #17795

elpaso committed:
Indentation

elpaso committed:
Indentation 2

jef-n committed:
debian packaging: update dependency

borysiasty committed:
[tr] Polish translation update

jef-n committed:
german translation update

stevenmizuno committed:
Qt4 doesn't have QStringLiteral

jef-n committed:
debian packaging: add bionic

jef-n committed:
avoid including GDAL C++ api (fixes #17849)

alexbruy committed:
remove duplicated parameter from r.in.wms module (fix #17815)

(cherry picked from commit b8518aef9cb7a514cc2c98a5b24e90cd27658a33)

alexbruy committed:
[processing] update parameter name for v.segments (fix #17850)

radosuav committed:
Merge remote-tracking branch 'qgis/release-2_18' into release-2_18

radosuav commented:
Superseded by #6080 due to git confusion.
